src/oidc_child/oidc_child_util.h

```diff
@@ -0,0 +1,96 @@
+/*
+    SSSD
+
+    Helper child for OIDC and OAuth 2.0 Device Authorization Grant
+
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2022 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __OIDC_CHILD_UTIL_H__
+#define __OIDC_CHILD_UTIL_H__
+
+#include <jansson.h>
+#include "util/util.h"
+
+struct devicecode_ctx {
+    bool libcurl_debug;
+    char *ca_db;
+    char *device_authorization_endpoint;
+    char *token_endpoint;
+    char *userinfo_endpoint;
+    char *jwks_uri;
+    char *scope;
```

Comment:
Looks like all `char *` above could be `const char *`?

Reply:
Perhaps not worth doing so, taking into account `sss_erase_talloc_mem_securely` d-tor attached...

Reply:
fixed